clustered/server/src/main/java/org/ehcache/clustered/server/offheap/StorageEngineExtensions.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.offheap;
+
+public interface StorageEngineExtensions {
+  default void chainFreed(long chainAddress) {}
+  default void chainModified(long chainAddress) {}
+  default void chainMoved(long fromChainAddress, long toChainAddress) {}
+
+  OffHeapChainExtensionHeader getExtendedHeader();
```

Comment:
If we eliminate this interface (as described in other comments) then this method goes, and therefore this interface can go too.